src/r_main.h

```diff
@@ -125,6 +115,39 @@ void R_InterpolateTextureOffsets (void);
 
 void R_InitAnyRes(void);
 
+// [AM] Fractional part of the current tic, in the half-open
+//      range of [0.0, 1.0).  Used for interpolation.
+extern fixed_t fractionaltic;
+
+inline static fixed_t LerpFixed(fixed_t oldvalue, fixed_t newvalue)
+{
+    return (oldvalue + FixedMul(newvalue - oldvalue, fractionaltic));
+}
+
+// [AM] Interpolate between two angles.
+inline static angle_t LerpAngle(angle_t oangle, angle_t nangle)
+{
+    if (nangle == oangle)
+        return nangle;
+    else if (nangle > oangle)
+    {
+        if (nangle - oangle < ANG270)
+            return oangle + (angle_t)((nangle - oangle) * FIXED2DOUBLE(fractionaltic));
```

Comment:
Will it be faster to use e.g. `lrintf()` or `lround()` here instead of casting to `(angle_t)`?

Reply:
Isn't every part of the renderer performance critical? :neckbeard:

Reply:
> Isn't every part of the renderer performance critical?

I think the critical part is most of `r_draw.c` and the algorithms in `r_bsp.c` etc. Woof spends most of the CPU time on the `R_DrawColumn` function. But sure, we could just use `lrint`, we already use `lround` anyway.

Reply:
Just built it with `lrint` and `/fp:fast`, got no difference in timedemo performance. So not sure if it is worth the risk.
<details>

```diff
diff --git a/cmake/WoofSettings.cmake b/cmake/WoofSettings.cmake
index 9eac493..953c727 100644
--- a/cmake/WoofSettings.cmake
+++ b/cmake/WoofSettings.cmake
@@ -107,6 +107,15 @@ if(ENABLE_HARDENING)
     _checked_add_link_option(-Wl,-z,relro)
 endif()
 
+option(ENABLE_FASTMATH "Enable fast float point math" ON)
+if(ENABLE_FASTMATH)
+    if(MSVC)
+        _checked_add_compile_option(/fp:fast)
+    else()
+        _checked_add_compile_option(-ffast-math)
+    endif()
+endif()
+
 include(CheckIPOSupported)
 check_ipo_supported(RESULT HAVE_LTO)
 
diff --git a/src/r_main.h b/src/r_main.h
index 79a0843..aa4a37a 100644
--- a/src/r_main.h
+++ b/src/r_main.h
@@ -118,7 +118,7 @@ extern fixed_t fractionaltic;
 
 inline static fixed_t LerpFixed(fixed_t oldvalue, fixed_t newvalue)
 {
-    return (oldvalue + FixedMul(newvalue - oldvalue, fractionaltic));
+    return oldvalue + FixedMul(newvalue - oldvalue, fractionaltic);
 }
 
 // [AM] Interpolate between two angles.
@@ -129,16 +129,16 @@ inline static angle_t LerpAngle(angle_t oangle, angle_t nangle)
     else if (nangle > oangle)
     {
         if (nangle - oangle < ANG270)
-            return oangle + (angle_t)((nangle - oangle) * FIXED2DOUBLE(fractionaltic));
+            return oangle + lrint((nangle - oangle) * FIXED2DOUBLE(fractionaltic));
         else // Wrapped around
-            return oangle - (angle_t)((oangle - nangle) * FIXED2DOUBLE(fractionaltic));
+            return oangle - lrint((oangle - nangle) * FIXED2DOUBLE(fractionaltic));
     }
     else // nangle < oangle
     {
         if (oangle - nangle < ANG270)
-            return oangle - (angle_t)((oangle - nangle) * FIXED2DOUBLE(fractionaltic));
+            return oangle - lrint((oangle - nangle) * FIXED2DOUBLE(fractionaltic));
         else // Wrapped around
-            return oangle + (angle_t)((nangle - oangle) * FIXED2DOUBLE(fractionaltic));
+            return oangle + lrint((nangle - oangle) * FIXED2DOUBLE(fractionaltic));
     }
 }
 ```
</details>

Reply:
Okay, wow, thanks for checking! One could also argue why we use `lround()` instead of `lrint()`, but if even completely omitting these functions makes no difference in this case, I don't think we will notice a difference in such implementation details.

Reply:
`lrint()` can use different rounding modes, so `lround` might be clearer. Anyway, we have very few floating-point calculations, so I don't think we should worry about it.

Reply:
@fabiangreffrath Should we merge this?

Reply:
Sorry, I didn't realize I haven't approved this yet.